### PR TITLE
Gradle

### PR DESCRIPTION
**Pull request by BrainStone**
Changed build system from cmake to gradle

(Pull #2 first I'd say)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,4 +35,344 @@
 *.out
 *.app
 
-.idea/*
```
Comment (fivunlm) [inline .gitignore:31]: Are these removed as part of another wildcard? or we should add it?

**Review by fivunlm — CHANGES_REQUESTED** [review]
Great work, just a minor comment on ignore file

File: .gitignore (hunk 1 above)
Comment (BrainStone) [reply .gitignore:31]: This line is added back further down.

**Review by fivunlm — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (fivunlm) [reply .gitignore:31]: You're right I've missed that

**Pull request merged**